### PR TITLE
feat: Implement static, variable and template catalog field mapping

### PR DESCRIPTION
**Pull request by karthik-sivadas**
## Description
feat: Implement static, variable and template catalog field mapping

## Related Issue
None

## Type of Change
- [x] Feature
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would impact existing functionality)
- [ ] Documentation update
- [ ] Chore
  
## How Has This Been Tested?
Wrote unit test and tested the API locally

## Checklist:
- [x] Ensure a branch name is prefixed with `feature`, `bugfix`, `hotfix`, `release` or `chore` followed by `/` and branch name e.g `feature/add-salesforce-connector`

### CODE REVIEW

File: spec/lib/reverse_etl/transformers/user_mapping_spec.rb [hunk 1]
```diff
@@ -50,5 +52,112 @@
...
+              "template_field" => "Transformed 40"
+            }
+          }
+        }
+
+        expect(results).to eq(expected_result)
+      end
+    end
+
+    context "when using standard, static, and template mapping" do
```
Comment (subintp) [inline spec/lib/reverse_etl/transformers/user_mapping_spec.rb:82]: Duplicate tests ?

File: spec/lib/reverse_etl/transformers/user_mapping_spec.rb [hunk 2]
```diff
@@ -50,5 +52,112 @@
...
+      let(:mapping) do
+        [
+          { mapping_type: "standard", from: "cr_item_sk", to: "id" },
+          { mapping_type: "static", to: "attributes.properties.static_field", value: "static_value" },
+          { mapping_type: "template", to: "attributes.properties.template_field",
+            template: "Transformed {{cr_reason_sk}}" }
+        ]
+      end
+
+      it "transforms record according to v2 mappings" do
```
Comment (subintp) [inline spec/lib/reverse_etl/transformers/user_mapping_spec.rb:92]: Duplicate tests ?

**Review by subintp — CHANGES_REQUESTED** [review]

File: app/interactors/syncs/sync_configurations.rb [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        }
+      }
+    end
+
+    def template_variables
+      {
+        current_timestamp: {
+          type: "datetime",
+          description: "The current timestamp as an ISO 8601 string YYYY-MM-DDTHH:mm:ss.sssZ.",
+          value: "{{ 'now' | date: '%Y-%m-%dT%H:%M:%S.%L%z' }}"
```
[the remark this answers is not in the document]
Comment (karthik-sivadas) [reply app/interactors/syncs/sync_configurations.rb:59]: This is just a template users can change. This is assigned as a placeholder value which will be placed in the text box which the user can change based on the users need.

File: app/interactors/syncs/sync_configurations.rb [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        includes: {
+          description: "Check input for a substring. This function can be used for control flow.",
+          value: "{{ includes : '<substring>' }}"
+        },
+        regex_replace: {
+          description: "Search and replace substrings of input using RegEx",
+          value: "{{ regex_replace : '<[a-zA-Z]>', '<replacement>', '<flags>' }}"
+        },
+        regex_test: {
+          description: "Check input for a RegEx match. Row will be rejected if no match is found",
```
[the remark this answers is not in the document]
Comment (karthik-sivadas) [reply app/interactors/syncs/sync_configurations.rb:79]: The current implementation does not, will update accordingly.

File: app/interactors/syncs/sync_configurations.rb [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+          description: "Cast input to specified type. Supported values: string, number, boolean",
+          value: "{{ cast: '<string>' }}"
+        },
+        includes: {
+          description: "Check input for a substring. This function can be used for control flow.",
+          value: "{{ includes : '<substring>' }}"
+        },
+        regex_replace: {
+          description: "Search and replace substrings of input using RegEx",
+          value: "{{ regex_replace : '<[a-zA-Z]>', '<replacement>', '<flags>' }}"
```
[the remark this answers is not in the document]
Comment (karthik-sivadas) [reply app/interactors/syncs/sync_configurations.rb:76]: https://javascript.info/regexp-introduction

File: lib/liquid/custom_filters.rb [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    def includes(input, substring)
+      input.include?(substring)
+    end
+
+    def regex_replace(input, pattern, replacement = "", flags = "")
+      re = build_regexp(pattern, flags)
+      input.gsub(re, replacement)
+    end
+
+    def regex_test(input, pattern, _replacement = "", flags = "")
```
Comment (afthabvp) [inline lib/liquid/custom_filters.rb:25]: regex_test=> choose be the name

Comment (afthabvp) [inline lib/liquid/custom_filters.rb:25]: why _replacement?

File: app/interactors/syncs/sync_configurations.rb [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+      {
+        data: {
+          configurations: {
+            catalog_mapping_types:
+          }
+        }
+      }
+    end
+
+    def catalog_mapping_types
```
[the remark this answers is not in the document]
Comment (karthik-sivadas) [reply app/interactors/syncs/sync_configurations.rb:23]: Done

File: app/models/sync.rb [hunk 8]
```diff
@@ -90,4 +91,14 @@ def schedule_sync
   rescue StandardError => e
     Rails.logger.error "Failed to schedule sync with Temporal. Error: #{e.message}"
   end
+
+  def stream_name_exists?
```
[the remark this answers is not in the document]
Comment (karthik-sivadas) [reply app/models/sync.rb:95]: done

**Review by subintp — APPROVED** [review]

**Review by afthabvp — APPROVED** [review]

**Pull request merged**